ara/callback/log_ara.py

```diff
@@ -121,6 +139,10 @@ def log_task(self, result, status, **kwargs):
 
         db.session.add(self.taskresult)
 
+        if self.task.action == 'setup':
+            values = json.dumps(result._result['ansible_facts'])
+            self.create_or_update_facts(host, values)
```

Comment:
We don't need a create_or_update function here.  See my comment on `models.py`; if we have a one-to-one relationship, we can just assign `host.facts = json.dumps(result._result['ansible_facts'])`.  It doesn't matter of there is an existing value or not; it will get replaced.

That means we can get rid of the entire `create_or_update_facts` method.